pkg/inputs/snmp/disco.go

```diff
@@ -248,7 +276,7 @@ func doubleCheckHost(result scan.Result, timeout time.Duration, ctl chan bool, m
 	foundDevices[result.Host.String()] = &device
 }
 
-func addDevices(foundDevices map[string]*kt.SnmpDeviceConfig, snmpFile string, conf *kt.SnmpConfig, isTest bool, log logger.ContextL) error {
+func addDevices(ctx context.Context, foundDevices map[string]*kt.SnmpDeviceConfig, snmpFile string, conf *kt.SnmpConfig, isTest bool, log logger.ContextL) (error, int, int, int) {
```

Comment:
Should we do something like create a struct for the return?  Something like
```type snmpDevices {
      added int
      replaced int
      delta int
}
```
and then return `(*snmpDevices, error)`?  It makes it a bit more readable IMO.  I know I'll forget this order down the road ;)

Reply:
hah, totally.